filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobPathBuilderFactory.scala

```diff
@@ -5,37 +5,49 @@ import com.typesafe.config.Config
 final case class StorageAccountName(value: String) {override def toString: String = value}
 final case class EndpointURL(value: String) {override def toString: String = value}
-final case class WorkspaceId(value: String) {override def toString: String = value}
+final case class WorkspaceId(value: UUID) {override def toString: String = value.toString}
+final case class ContainerResourceId(value: UUID) {override def toString: String = value.toString}
 final case class WorkspaceManagerURL(value: String) {override def toString: String = value}
-final case class BlobPathBuilderFactory(globalConfig: Config, instanceConfig: Config, singletonConfig: BlobFileSystemConfig) extends PathBuilderFactory {
-  val subscription: Option[SubscriptionId] = instanceConfig.as[Option[String]]("subscription").map(SubscriptionId)
-  val container: BlobContainerName = BlobContainerName(instanceConfig.as[String]("container"))
-  val endpoint: EndpointURL = EndpointURL(instanceConfig.as[String]("endpoint"))
-  val workspaceId: Option[WorkspaceId] = instanceConfig.as[Option[String]]("workspace-id").map(WorkspaceId)
-  val expiryBufferMinutes: Long = instanceConfig.as[Option[Long]]("expiry-buffer-minutes").getOrElse(10)
-  val workspaceManagerURL: Option[WorkspaceManagerURL] = singletonConfig.config.as[Option[String]]("workspace-manager-url").map(WorkspaceManagerURL)
-  val b2cToken: Option[String] = instanceConfig.as[Option[String]]("b2cToken")
-
-  val blobTokenGenerator: BlobTokenGenerator = (workspaceManagerURL, b2cToken, workspaceId) match {
-    case (Some(url), Some(token), Some(workspaceId)) =>
-      val wsmClient: WorkspaceManagerApiClientProvider = new HttpWorkspaceManagerClientProvider(url, token)
-      // parameterizing client instead of URL to make injecting mock client possible
-      BlobTokenGenerator.createBlobTokenGenerator(container, endpoint, workspaceId, wsmClient)
-    case _ =>
-      BlobTokenGenerator.createBlobTokenGenerator(container, endpoint, subscription)
-  }
 
-  val fsm: BlobFileSystemManager = BlobFileSystemManager(container, endpoint, expiryBufferMinutes, blobTokenGenerator)
+final case class BlobPathBuilderFactory(globalConfig: Config, instanceConfig: Config, singletonConfig: BlobFileSystemConfigWrapper) extends PathBuilderFactory {
+
+  private val config = singletonConfig.config
+  private val container = config.blobContainerName
+  private val endpoint = config.endpointURL
+  private val subscription = config.subscriptionId
+  private val expiryBufferMinutes = config.expiryBufferMinutes
+
+  /**
+   * This generator is responsible for producing a valid SAS token for use in accessing a Azure blob storage container
+   * Two types of generators can be produced here:
+   * > Workspace Manager (WSM) mediated SAS token generator, used to create SAS tokens that allow access for
+   * blob containers mediated by the WSM, and is enabled when a WSM config is provided.
```

Comment:
Can we add a note that this is how Cromwell operates in Terra, and native SAS token generation is intended for use outside Terra?